packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -496,6 +502,10 @@ export class GadgetConnection {
 
       const retryOnClose = (event: unknown) => {
         if (isCloseEvent(event)) {
+          if (event.code == 4429) {
+            return wrappedReject(new GadgetTooManyRequestsError(event.reason));
```

Comment:
I think we want to do like below and reset listeners:

```
resetListeners()
return wrappedReject(new GadgetTooManyRequestsError(event.reason));
```

Reply:
good point, makes more sense to just clearListeners.